packages/uma-auth-client/src/components/UmaConnectButton.tsx

```diff
@@ -48,9 +48,12 @@ const UmaConnectButton = (props: Props) => {
     }
 
     let shouldOpenModalImmediately: boolean = false;
-    if (codeVerifier && !isConnected && step !== Step.WaitingForApproval) {
+    if (codeVerifier && !isConnected && step !== Step.WaitingForApproval && step !== Step.ErrorConnecting) {
```

Comment:
@bsiaotickchong I'm wondering if this should be looking for explicitly only certain steps rather than !== to these two? I could see endless loop situations in some cases.

Reply:
Hm I see what you mean. I guess if we set it look for "Connect" that would work, but I think it could be confusing because it's not clear if we change the initial step (from Connect to something else) that this will break. Maybe the best solution is to create a more explicit state machine that defines what step changes are possible